docs/bucket/replication/README.md

```diff
@@ -18,8 +18,60 @@ mc admin bucket remote add myminio/srcbucket https://accessKey:secretKey@replica
 Role ARN = 'arn:minio:replication:us-east-1:c5be6b16-769d-432a-9ef1-4567081f3566:destbucket'
 ```
 
-Note that the admin needs *s3:GetReplicationConfigurationAction* permission on source cluster. The credential used at the destination requires *s3:ReplicateObject* permission. Once successfully created and authorized this generates a replication target ARN.  The command below lists all the currently authorized replication targets:
+> NOTE:  The user running the above command needs *s3:GetReplicationConfiguration* and *s3:GetBucketVersioning* permission on the source cluster (we do not recommend running root credentials/super admin with replication). The access credentials used at the destination requires *s3:ReplicateObject* permission.
 
+Minimal policy on source bucket should have following permissions on the source bucket.
+```
+{
+ "Version": "2012-10-17",
+ "Statement": [
+  {
+   "Effect": "Allow",
+   "Action": [
+    "s3:GetReplicationConfiguration",
+    "s3:ListBucket",
+    "s3:GetBucketLocation",
+    "s3:GetBucketVersioning"
+   ],
+   "Resource": [
+    "arn:aws:s3:::srcbucket"
+   ]
+  }
+}
+```
+The access key configured to perform replication on the target cluster should have minimal permissions as in example below.
+```
+{
+ "Version": "2012-10-17",
+ "Statement": [
+  {
+   "Effect": "Allow",
+   "Action": [
+    "s3:GetBucketVersioning"
+   ],
+   "Resource": [
+    "arn:aws:s3:::destbucket"
+   ]
+  },
+  {
+   "Effect": "Allow",
+   "Action": [
+    "s3:ReplicateTags",
+    "s3:GetObject",
+    "s3:GetObjectVersion",
+    "s3:GetObjectVersionTagging",
+    "s3:PutObject",
+    "s3:ReplicateObject"
```

Comment:
Do we need these permissions?

"s3:PutObjectRetention",
"s3:PutObjectLegalHold",
"s3:PutObjectTagging",

Reply:
only if object locking is enabled.